framework/src/utils/MooseObjectParameterName.C

```diff
@@ -0,0 +1,126 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+// MOOSE includes
+#include "MooseObjectParameterName.h"
```

Comment:
It seems like there is some relationship between `MooseObjectName` and `MooseObjectParameterName`. I see very similar parsing technics in these two unrelated classes. If you really want two, can one of them inherit most of it functionality from the other?